cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -802,19 +834,39 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
 
         Opcode::Bnot => {
             let ty = ty.unwrap();
+            let size = ty.bytes() as u8;
+            let src = put_input_in_reg(ctx, inputs[0]);
+            let dst = get_output_reg(ctx, outputs[0]);
+            ctx.emit(Inst::gen_move(dst, src, ty));
+
             if ty.is_vector() {
-                unimplemented!("vector bnot");
+                let tmp = ctx.alloc_tmp(RegClass::V128, ty);
+                ctx.emit(Inst::equals(ty, RegMem::from(tmp), tmp));
+                ctx.emit(Inst::xor(ty, RegMem::from(tmp), dst));
             } else if ty.is_bool() {
                 unimplemented!("bool bnot")
             } else {
-                let size = ty.bytes() as u8;
-                let src = put_input_in_reg(ctx, inputs[0]);
-                let dst = get_output_reg(ctx, outputs[0]);
-                ctx.emit(Inst::gen_move(dst, src, ty));
                 ctx.emit(Inst::not(size, dst));
             }
         }
 
+        Opcode::Bitselect => {
+            let ty = ty.unwrap();
+            let c = put_input_in_reg(ctx, inputs[0]);
+            let x = put_input_in_reg(ctx, inputs[1]);
+            let y = put_input_in_reg(ctx, inputs[2]);
+            let dst = get_output_reg(ctx, outputs[0]);
+
+            if ty.is_vector() {
+                ctx.emit(Inst::and(ty, RegMem::reg(c.clone()), Writable::from_reg(x)));
```

Comment:
Here (and below for the condition too): it's incorrect to get a writable register for input values. If the input vreg is used in more vcode instructions after this one, then its value will be clobbered by this particular vcode inst (and resulting machine code). We'll need temporaries here instead.

Reply:
Ok, I think I did that correctly... I ported some more CLIF filetests to ensure no extra moves are required.